### ISSUE TITLE
setting IFS to "" crashes fish on some completions

### ISSUE THREAD

**Issue by shalva97** [fish-shell/fish-shell#5398]
fish --version 2.7.1
os: Linux archLinux 4.19.4-arch1-1-ARCH x86_64

setting IFS to empty string and then autocompleting last character of any executable crashes terminal

```
~/p/s/sdfhyi ❯❯❯ ls
asdf*  asdf.go
~/p/s/sdfhyi ❯❯❯ sh -c 'env HOME=$(mktemp -d) fish'
/h/s/p/s/sdfhyi ❯❯❯ ls
asdf*  asdf.go
/h/s/p/s/sdfhyi ❯❯❯ set IFS ""
/h/s/p/s/sdfhyi ❯❯❯ ./asdterminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::basic_string: __pos (which is 3) > this->size() (which is 0)
fish: “sh -c 'env HOME=$(mktemp -d) fi…” terminated by signal SIGABRT (Abort)
~/p/s/sdfhyi ❯❯❯
```

**Comment by faho** [comment]
I can't reproduce this, also on arch, with their fish 2.7.1 or my self-built 2.7.1.r1757.gbd1ceefb1.

**Comment by faho** [comment]
Okay, no, I managed to reproduce, with 2.7.1. I needed an actual "asdf" and "asdf.go" (with the "asdf" being a copy of my "ls", and set executable). Then enter ./asd<TAB>.

The backtrace looks like:

```
Stack trace of thread 7749:
                #0  0x00007fb84a99ad7f raise (libc.so.6)
                #1  0x00007fb84a985672 abort (libc.so.6)
                #2  0x00007fb84abb058e _ZN9__gnu_cxx27__verbose_terminate_handlerEv (libstdc++.so.6)
                #3  0x00007fb84abb6dfa _ZN10__cxxabiv111__terminateEPFvvE (libstdc++.so.6)
                #4  0x00007fb84abb6e57 _ZSt9terminatev (libstdc++.so.6)
                #5  0x00007fb84abb70ac __cxa_throw (libstdc++.so.6)
                #6  0x00007fb84abb27aa _ZSt24__throw_out_of_range_fmtPKcz (libstdc++.so.6)
                #7  0x000055c76ae30ece _ZN11completer_t17complete_cmd_descERKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEE (fish)
                #8  0x000055c76ae31126 n/a (fish)
                #9  0x000055c76ae318f8 _Z8completeRKNSt7__cxx1112basic_stringIwSt11char_traitsIwESaIwEEEPSt6vectorI12completion_tSaIS8_EEjRK19env_vars_>
                #10 0x000055c76aea64a6 _Z15reader_readlinei (fish)
                #11 0x000055c76aea8280 _Z11reader_readiRK10io_chain_t (fish)
                #12 0x000055c76adf78f6 main (fish)
                #13 0x00007fb84a987223 __libc_start_main (libc.so.6)
                #14 0x000055c76adf7aaa _start (fish)
```

As it turns out, we've fixed a crash in complete_cmd_desc - #5014. Since I can't reproduce this with git master, I'm going to assume it's a duplicate of that.
